src/main/java/com/tabnine/balloon/CompletionHintTooltip.java

```diff
@@ -0,0 +1,51 @@
+package com.tabnine.balloon;
+
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.editor.Editor;
+import com.tabnine.selections.CompletionListener;
+import com.tabnine.selections.CompletionObserver;
+import com.tabnine.state.UserState;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+public class CompletionHintTooltip {
+  private static final GotItTooltip completionHintTooltip =
+      new GotItTooltip(
+          "first-completion-hint",
+          "Your first completion from Tabnine",
+          "Press <b>Tab</b> to accept Tabnine's suggestion",
+          () -> UserState.getInstance().getCompletionHintState().setIsShown(true));
+
+  public static void handle(Editor editor) {
+    try {
+      UserState userState = UserState.getInstance();
+      if (userState.getCompletionHintState().isEligibleForCompletionHint()) {
+        if (!completionHintTooltip.isVisible()) {
```

Comment:
invert conditions to something like
```java
if (!userState.getCompletionHintState().isEligibleForCompletionHint() || 
    completionHintTooltip.isVisible()) {
    // "else" code here
    return;
}
// main code here
```

Reply:
in this specific example, I want to handle the else case of the isEligible condition, and ignore the else case of the isVisible

Reply:
something like
`      if (!userState.getCompletionHintState().isEligibleForCompletionHint()) {
        userState.getCompletionHintState().setIsShown(true);
        return;
      }
      if (!completionHintTooltip.isVisible()) {
        return;
      }`